integration/tests/bft/signature-verification/signature_verification_test.go

```diff
@@ -0,0 +1,37 @@
+package signature_verification
+
+import (
+	"fmt"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/require"
+	"github.com/stretchr/testify/suite"
+
+	"github.com/onflow/flow-go/utils/unittest"
+)
+
+type SignatureVerificationTestSuite struct {
+	Suite
+}
+
+func TestSignatureVerification(t *testing.T) {
+	suite.Run(t, new(SignatureVerificationTestSuite))
+}
+
+// TestSignatureVerificationE2E ensures that the libp2p signature verification is working as expected.
+// The test network is set up with 3 corrupted AN, 2 attackers and one victim. 1 attacker AN that has message signing disabled, this
+// node will be used to send messages without signatures to the victim AN. These messages should never be
+// delivered to the victim as the signature verification is expected to fail. 1 attacker AN with message
+// signing enabled, this node will send messages that are expected to be delivered to the victim AN.
+func (s *SignatureVerificationTestSuite) TestSignatureVerificationE2E() {
+	s.Orchestrator.sendUnauthorizedMsgs(s.T())
+	s.Orchestrator.sendAuthorizedMsgs(s.T())
+	unittest.RequireReturnsBefore(s.T(), s.Orchestrator.authorizedEventReceivedWg.Wait, 5*time.Second, "could not send authorized messages on time")
+
+	// messages without correct signature should have all been rejected at the libp2p level and not be delivered to the victim AN.
+	require.Equal(s.T(), int64(0), s.Orchestrator.unauthorizedEventsReceived.Load(), fmt.Sprintf("expected to not receive any unauthorized messages instead got: %d", s.Orchestrator.unauthorizedEventsReceived.Load()))
+
+	// messages with correct message signatures are expected to always pass libp2p signature verification and be delivered to the victim AN.
+	require.Equal(s.T(), int64(numOfAuthorizedEvents), s.Orchestrator.authorizedEventsReceived.Load(), fmt.Sprintf("expected to receive %d authorized events got: %d", numOfAuthorizedEvents, s.Orchestrator.unauthorizedEventsReceived.Load()))
+}
```

Comment:
Please also add a check to evaluate the content of the authorized events sent, i.e., we desire all the sent authorized events to be received intact.

Reply:
https://github.com/onflow/flow-go/pull/3689/commits/516319f75db223f7d7f76faf16420905dd41ad6c